src/main/java/duke/Parser.java

```diff
@@ -0,0 +1,207 @@
+package duke;
+
+import command.Command;
+import command.DeadlineCommand;
+import command.DeleteCommand;
+import command.EmptyCommand;
+import command.EventCommand;
+import command.ExitCommand;
+import command.FindCommand;
+import command.ListCommand;
+import command.MarkCommand;
+import command.TodoCommand;
+import command.UnmarkCommand;
+import task.Deadline;
+import task.Event;
+import task.Todo;
+
+import java.time.LocalDate;
+import java.time.format.DateTimeParseException;
+import java.util.Scanner;
+
+/**
+ * Represents a parser class that will be in charge of parsing user commands
+ * for Duke.
+ */
+public class Parser {
+    private static final String FLAG_BYE = "bye";
+    private static final String FLAG_LIST = "list";
+    private static final String FLAG_MARK = "mark";
+    private static final String FLAG_UNMARK = "unmark";
+    private static final String FLAG_TODO = "todo";
+    private static final String FLAG_DEADLINE = "deadline";
+    private static final String FLAG_EVENT = "event";
+    private static final String FLAG_DELETE = "delete";
+    private static final String FLAG_FIND = "find";
+
+    /**
+     * Parses the input passed in and return the respective command.
+     *
+     * @param input The user input.
+     * @return The command represented by input.
+     * @throws DukeException If the command format is invalid or unrecognized.
+     */
+    public static Command parseCommand(String input) throws DukeException {
+        input = input.trim();
+
+        // skip past empty lines
+        if (input.isEmpty()) {
+            return new EmptyCommand();
+        }
+
+        String[] parts = input.split("\\s+", 2);
+        String command = parts[0];
+        String value = parts.length >= 2 ? parts[1].trim() : "";
+
+        if (command.equals(FLAG_BYE)) {
+            if (!value.isEmpty()) {
+                throw new DukeException("Oops!!! The bye command should not be followed by any description");
+            }
+            return new ExitCommand();
+        }
+
+        if (command.equals(FLAG_LIST)) {
+            if (!value.isEmpty()) {
+                throw new DukeException("Oops!!! The list command should not be followed by any description");
+            }
+            return new ListCommand();
+        }
+
+        if (command.equals(FLAG_MARK)) {
+            Scanner tempSc = new Scanner(value);
+
+            if (!tempSc.hasNextInt()) {
+                tempSc.close();
+                throw new DukeException("Oops!!! Invalid argument of a mark command");
+            }
+
+            int index = tempSc.nextInt();
+            if (tempSc.hasNext()) {
+                // means invalid formatting for done command
+                tempSc.close();
+                throw new DukeException("Oops!!! Invalid argument of a mark command");
+            }
+            tempSc.close();
```

Comment:
May want to abstract out the code handling input and exceptions (Don't repeat yourself)